docs/en_us/platform_api/source/courses/blocks.rst

```diff
@@ -0,0 +1,350 @@
+.. _Courses API Blocks Resource:
+
+########################################
+Courses API Blocks Resource
+########################################
+
+With the Courses API **Blocks** resource, you can complete the
+following tasks.
+
+
+.. contents::
+   :local:
+   :depth: 1
+
+.. _Get a list of the course blocks in a course:
+
+*******************************************************
+Get a List of Course Blocks in a Specified Course
+*******************************************************
+
+The endpoint to get a list of course blocks in a specified block tree
+``/api/courses/v1/blocks/?course_id=<course_id>``.
+
+=====================
+Use Case
+=====================
+
+Get a list of course blocks in a specified course. Response results depend on
+the requesting user's access level.
+
+=====================
+Example Request
+=====================
+
+GET /api/courses/v1/blocks/?course_id=<course_id>
+
+GET /api/courses/v1/blocks/?course_id=edX%2FDemoX%2FDemo_Course
+&all_blocks=true&requested_fields=graded,format,student_view_multi_device
+
+=====================
+Parameters
+=====================
+
+This view redirects to /api/courses/v1/blocks/<root_usage_key>/ for the root
+block usage key of the course specified by course_id.
+
+This view accepts the following parameters.
+
+* block_counts: to return the aggregate number of blocks (including child
+  blocks) of each type specified in the "block_counts" response value.
+  Example: ``block_counts=video,problem``.
+
+* block_types_filter: Specify the types of blocks to be included in results.
+  Possible values are: ``sequential``, ``vertical``, ``html``, ``problem``,
+  ``video``   and ``discussion``. Example: ``block_types_filter=problem,html``.
+
+* course_id: (string, required) The ID of the course whose block data you are
```

Comment:
It looks like the course ID needs to be URL encoded. A string like course-v1:edX+DemoX+Demo_Course will be rejected. But course-v1%3AedX%2BDemoX%2BDemo_Course is accepted.